library/src/main/java/software/amazon/location/auth/LocationCredentialsProvider.kt

```diff
@@ -83,13 +79,39 @@ class LocationCredentialsProvider {
     }
 
     /**
-     * Retrieves the Cognito credentials provider.
-     * @return The CognitoCachingCredentialsProvider instance.
+     * Generates and stores AWS credentials.
+     *
+     * This function fetches the identity pool ID and region from the key-value store,
+     * uses these to retrieve an identity ID and credentials from AWS Cognito, and then
+     * stores these credentials using a CognitoCredentialsProvider.
+     *
+     * @return True if the credentials were successfully generated and stored, false otherwise.
+     */
+    suspend fun generateCredentials(): Boolean {
+        val identityPoolId = awsKeyValueStore.get(IDENTITY_POOL_ID)
+        val region = awsKeyValueStore.get(REGION)
+        val cognitoCredentialsHttpHelper = CognitoCredentialsClient(region)
```

Comment:
can these values be null here? if so, throw an exception

Reply:
Done